code/handlers/dotz.q

```diff
@@ -26,17 +26,46 @@ if[not@[value;`SAVED.ORIG;0b]; / onetime save only
     txtc:txt[neg 60-last system"c"];txtC:txt[neg 60-last system"C"];
     pzlist:` sv'`.z,'`pw`po`pc`pg`ps`pi`ph`pp`ws`exit;
     .dotz.undef:pzlist where not @[{not(::)~value x};;0b] each pzlist;
-    .dotz.pw.ORIG:.z.pw:@[.:;`.z.pw;{{[x;y]1b}}];
-    .dotz.po.ORIG:.z.po:@[.:;`.z.po;{;}];
-    .dotz.pc.ORIG:.z.pc:@[.:;`.z.pc;{;}];
-    .dotz.wo.ORIG:.z.wo:@[.:;`.z.wo;{;}];
-    .dotz.wc.ORIG:.z.wc:@[.:;`.z.wc;{;}];
-    .dotz.exit.ORIG:.z.exit:@[.:;`.z.exit;{;}];
-    .dotz.pg.ORIG:.z.pg:@[.:;`.z.pg;{.:}];
-    .dotz.ps.ORIG:.z.ps:@[.:;`.z.ps;{.:}];
-    .dotz.pi.ORIG:.z.pi:@[.:;`.z.pi;{{.Q.s value x}}];
+    .dotz.pw.ORIG:.dotz.set[`.z.pw;@[.:;`.z.pw;{{[x;y]1b}}]];
+    .dotz.po.ORIG:.dotz.set[`.z.po;@[.:;`.z.po;{;}]];
+    .dotz.pc.ORIG:.dotz.set[`.z.pc;@[.:;`.z.pc;{;}]];
+    .dotz.wo.ORIG:.dotz.set[`.z.wo;@[.:;`.z.wo;{;}]];
+    .dotz.wc.ORIG:.dotz.set[`.z.wc;@[.:;`.z.wc;{;}]];
+    .dotz.exit.ORIG:.dotz.set[`.z.exit;@[.:;`.z.exit;{;}]];
+    .dotz.pg.ORIG:.dotz.set[`.z.pg;@[.:;`.z.pg;{.:}]];
+    .dotz.ps.ORIG:.dotz.set[`.z.ps;@[.:;`.z.ps;{.:}]];
+    .dotz.pi.ORIG:.dotz.set[`.z.pi;@[.:;`.z.pi;{{.Q.s value x}}]];
```

Comment:
does `.dotz.set` return the definition? if not, these `*.ORIG` variables might not be set correctly?

could do e.g.

```q
.dotz.set[`.z.pw;.dotz.pw.ORIG:@[.:;`.z.pw;{{[x;y]1b}}]];
```

etc., or change `.dotz.set` to return the definition after setting & leave these lines as-is